### PR TITLE
feat: Enable configuration of resource tracking method

### LINKED ISSUE #575: Set application.resourceTrackingMethod via ArgoCD CRD
Enable the operator / ArgoCD CRD to set application.resourceTrackingMethod - see https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/#choosing-a-tracking-method

### PR DESCRIPTION
**Pull request by jannfis**
**What type of PR is this?**

/kind enhancement

**What does this PR do / why we need it**:

This allows the user to set the [resource tracking method](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_tracking/#choosing-a-tracking-method) used by Argo CD to identify managed resources. 

The setting for this is `.spec.resourceTrackingMethod`.

Valid values are `label`, `annotation` and `annotation+label`.

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #575 

**How to test changes / Special notes to the reviewer**:

1. Create default `ArgoCD` operand in your cluster without `resourceTrackingMethod` set
2. Check in `argocd-cm` that `application.resourceTrackingMethod` has been set to the default `label`
3. Update `ArgoCD` to set `.spec.resourceTrackingMethod` to `annotation+label`
4. Check in `argocd-cm` that `application.resourceTrackingMethod` has been set to `annotation+label` after reconciliation completed

### CODE REVIEW

File: controllers/argocd/role_test.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func TestReconcileArgoCD_reconcileRole_for_new_namespace(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, expectedNumberOfRoles, len(dexRoles))
 	assert.Equal(t, expectedRoleNamespace, dexRoles[0].ObjectMeta.Namespace)
-	
+
```
Comment (iam-veeramalla) [inline controllers/argocd/role_test.go:67]: This change can be deleted.

File: controllers/argocd/configmap.go [hunk 2]
```diff
@@ -178,6 +178,13 @@ func getResourceInclusions(cr *argoprojv1a1.ArgoCD) string {
 	return re
 }
 
+// getResourceTrackingMethod will return the resource tracking method for the given ArgoCD.
+func getResourceTrackingMethod(cr *argoprojv1a1.ArgoCD) string {
+	rtm := argoprojv1a1.ParseResourceTrackingMethod(cr.Spec.ResourceTrackingMethod)
+	log.Info(fmt.Sprintf("Found %s as tracking method", cr.Spec.ResourceTrackingMethod))
```
Comment (jopit) [inline controllers/argocd/configmap.go:184]: Would it make sense to log an error if the tracking method was invalid?  Not sure the user would ever see it, though.

**Review by iam-veeramalla — APPROVED** [review]
/approve
/lgtm

File: controllers/argocd/configmap.go (hunk 2 above)
Comment (jannfis) [reply controllers/argocd/configmap.go:184]: Yeah, I agree that we should probably log it out, as this is good practice. Thanks!

Comment (jannfis) [reply controllers/argocd/configmap.go]: Done.

**Pull request merged**